include/SW_Domain.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SWDOMAIN_H
+#define SWDOMAIN_H
+
+#include "include/SW_datastructs.h"
+
```

Comment:
don't we need to wrap the SW_Domain header content in 
```
#ifdef __cplusplus
extern "C" {
#endif
...
#ifdef __cplusplus
}
#endif
```

to avoid C++ name mangling as we do in our other headers?

Reply:
Was there a reason you put the include inside the `#idfed __cplusplus ...`? Obviously, it works but all our other header files have the include statements before (e.g., following https://stackoverflow.com/questions/16087451/where-is-the-best-place-to-put-the-ifdef-cplusplus-extern-c-endif)